File: 392-is-subsequence/is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
       int i=0;
       if(s.length()<1)return true;
       for(char c : t.toCharArray()){
           if(s.charAt(i) == c){
               i++;
               if(i == s.length()){
                   return true;
               }
           }
       }
       return false;

    }
}

File: 392-is-subsequence/is-subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        l = 0
        r = 0
        while l < len(s) and r < len(t):
            if s[l] == t[r]:
                l += 1
            r += 1
        # If we've checked all characters in s, it's a subsequence
        return l == len(s)